File: fabric/src/main/kotlin/StreamParam.kt
package com.obecto.perper.fabric
import org.apache.ignite.binary.BinaryReader
import org.apache.ignite.binary.BinaryWriter
import org.apache.ignite.binary.Binarylizable

class StreamParam(
    var stream: String,
    var filter: Map<String, Any?>,
    var localToData: Boolean
) : Binarylizable {
    override fun writeBinary(writer: BinaryWriter) {
        throw RuntimeException("Did not expect call to writeBinary")
    }

    override fun readBinary(reader: BinaryReader) {
        filter = reader.readMap<String, Any?>("filter")
        stream = reader.readString("stream")
        localToData = reader.readBoolean("localToData")
    }
}

File: src/Perper.Protocol/Cache/StreamParam.cs
using Apache.Ignite.Core.Binary;

namespace Perper.Protocol.Cache
{
    public class StreamParam : IBinarizable
    {
        public string Stream { get; set; }
        public IBinaryObject? Filter { get; set; }

        public StreamParam() : this("", null)
        {
        }

        public StreamParam(string stream, IBinaryObject? filter)
        {
            Stream = stream;
            Filter = filter;
        }

        public void WriteBinary(IBinaryWriter writer)
        {
            writer.WriteString("stream", Stream);
            writer.WriteObject("filter", Filter);
        }

        public void ReadBinary(IBinaryReader reader)
        {
            Stream = reader.ReadString("stream");
            Filter = reader.ReadObject<IBinaryObject>("filter");
        }
    }
}